include/finufft/defs.h

```diff
@@ -212,10 +227,14 @@ template<typename T> struct type3params {
   T X3, C3, D3, h3, gam3; // z
 };
 
-typedef struct FINUFFT_PLAN_S { // the main plan object, fully C++
+struct FINUFFT_PLAN_S { // the main plan object, fully C++
+  // added to silence a warning
+  FINUFFT_PLAN_S()                                  = default;
+  FINUFFT_PLAN_S(const FINUFFT_PLAN_S &)            = delete;
+  FINUFFT_PLAN_S &operator=(const FINUFFT_PLAN_S &) = delete;
```

Comment:
I would add a comment on why deleting these. It makes sense to do so as it would break the allocations of the internal workspace.

Reply:
I did it because I wanted to get rid of the compiler warnings, but the lines don't actually do anything. These functions are deleted anyway, since the `unique_ptr` member does not support them.